pkg/deploy/generator/scripts/util-system.sh

```diff
@@ -0,0 +1,300 @@
+        return 0
+    fi
+
+    if [ "$role" == "rp" ]; then
+        local -r rp_certs_basedir="/etc/aro-rp"
+        mkdir -p "$rp_certs_basedir"
+        base64 -d <<<"$ADMINAPICABUNDLE" > "$rp_certs_basedir/admin-ca-bundle.pem"
+        if [[ -n "$ARMAPICABUNDLE" ]]; then
+        base64 -d <<<"$ARMAPICABUNDLE" > "$rp_certs_basedir/arm-ca-bundle.pem"
+        fi
+        chown -R 1000:1000 "$rp_certs_basedir"
+    fi
+
+    # setting MONITORING_GCS_AUTH_ID_TYPE=AuthKeyVault seems to have caused mdsd not
+    # to honour SSL_CERT_FILE any more, heaven only knows why.
+    local -r ssl_certs_basedir="/usr/lib/ssl/certs"
+    mkdir -p "$ssl_certs_basedir"
+    csplit -f "$ssl_certs_basedir/cert-" -b %03d.pem /etc/pki/tls/certs/ca-bundle.crt /^$/1 "{*}" 1>/dev/null
+    c_rehash "$ssl_certs_basedir"
+
+    # we leave clientId blank as long as only 1 managed identity assigned to vmss
+    # if we have more than 1, we will need to populate with clientId used for off-node scanning
+    # shellcheck disable=SC2034
+    local -r nodescan_agent_filename="/etc/default/vsa-nodescan-agent.config"
+    # shellcheck disable=SC2034
+    local -r nodescan_agent_file="{
+    \"Nice\": 19,
+    \"Timeout\": 10800,
+    \"ClientId\": \"\",
+    \"TenantId\": $AZURESECPACKVSATENANTID,
+    \"QualysStoreBaseUrl\": $AZURESECPACKQUALYSURL,
+    \"ProcessTimeout\": 300,
+    \"CommandDelay\": 0
+  }"
+
+    write_file nodescan_agent_filename nodescan_agent_file true
+}
+
+# run_azsecd_config_scan
+run_azsecd_config_scan() {
+    log "starting"
+
+    local -ar configs=(
+        "baseline"
+        "clamav"
+        "software"
+    )
+
+    log "Scanning configuration files with azsecd ${configs[*]}"
+    # shellcheck disable=SC2068
+    for scan in ${configs[@]}; do
+        log "Scanning config file $scan now"
+        /usr/local/bin/azsecd config -s "$scan" -d P1D
+    done
+}
+
+# create_required_dirs
+create_required_dirs() {
+    create_dirs=(
+        /var/log/journal
+        /var/lib/waagent/Microsoft.Azure.KeyVault.Store
+        # Does not exist on devProxyVMSS
+        /var/opt/microsoft/linuxmonagent
+    )
+
+    # shellcheck disable=SC2068
+    for d in ${create_dirs[@]}; do
+        log "Creating directory $d"
+        mkdir -p "$d" || abort "failed to create directory $d"
+    done
+}
+
+# create_podman_networks()
+# args:
+# 1) nets - nameref, associative array; Networks to be created
+#       Key is the network name, value is the subnet with cidr notation
+create_podman_networks() {
+    local -n nets="$1"
+    log "starting"
+
+    # shellcheck disable=SC2068
+    for n in ${!nets[@]}; do
+        log "Creating podman network \"$n\" with subnet \"${nets[$n]}\""
+        podman network \
+            create \
+            --subnet "${nets["$n"]}" \
+            "$n"
+    done
+}
+
+# firewalld_configure_backend
+firewalld_configure_backend() {
+    log "starting"
+
+    log "Changing firewalld backend to iptables"
```

Comment:
Which components need `iptables`? In the scripts, it doesn't look like we are using any `iptables` commands directly.

Reply:
We don't use direct `iptables` rules via `firewalld` here at all.
I have seen that `firewalld` doesn't have everything needed to use the `nftables` backend (even though the `nftables.rpm` is installed by default).  
When trying to perform any operation, this error is returned by `firewalld` and `firewall-cmd`.
```bash
internal:0:0-0: Error: Could not process rule: No such file or directory
```

I couldn't find exactly what that error means, but everything seems to work find with the `iptables` backend.